In [5]:
!pip install bs4

  Created wheel for bs4: filename=bs4-0.0.1-py3-none-any.whl size=1279 sha256=0d1d4c8b47e7356f5a109ad42559682050d878a91d6b0943e247a2474a1c1345
  Stored in directory: c:\users\sugandh\appdata\local\pip\cache\wheels\0a\9e\ba\20e5bbc1afef3a491f0b3bb74d508f99403aabe76eda2167ca
Successfully built bs4


In [56]:
import requests
from bs4 import BeautifulSoup
import pandas as pd

#imported required modules and functions
#Requesting weblink and 

URL = 'https://en.wikipedia.org/w/index.php?title=List_of_postal_codes_of_Canada:_M&oldid=1019189119#cite_note-statcan-3'
response = requests.get(URL)
soup = BeautifulSoup(response.text,'html.parser')
#print(soup.prettify())


to get table from html,we need to use find, to find table first and find_all, to find td, wherever it is coming and covert it into a list <br> we gave every column a name in the starting and started appending data into it

In [71]:
table_contents=[]
table=soup.find('table')
for row in table.findAll('td'):
    cell = {}
    if row.span.text=='Not assigned':
        pass
    else:
        cell['PostalCode'] = row.p.text[:3]
        cell['Borough'] = (row.span.text).split('(')[0]
        cell['Neighborhood'] = (((((row.span.text).split('(')[1]).strip(')')).replace(' /',',')).replace(')',' ')).strip(' ')
        table_contents.append(cell)

#print(table_contents)

To convert dict into dataframe

In [70]:
df = pd.DataFrame(table_contents)

df['Borough']=df['Borough'].replace({'Downtown TorontoStn A PO Boxes25 The Esplanade':'Downtown Toronto Stn A','East TorontoBusiness reply mail Processing Centre969 Eastern':'East Toronto Business','EtobicokeNorthwest':'Etobicoke Northwest','East YorkEast Toronto':'East York/East Toronto','MississaugaCanada Post Gateway Processing Centre':'Mississauga'})
df

PostalCode                Borough  \
0          M3A             North York   
1          M4A             North York   
2          M5A       Downtown Toronto   
3          M6A             North York   
4          M7A           Queen's Park   
..         ...                    ...   
98         M8X              Etobicoke   
99         M4Y       Downtown Toronto   
100        M7Y  East Toronto Business   
101        M8Y              Etobicoke   
102        M8Z              Etobicoke   

                                          Neighborhood  
0                                            Parkwoods  
1                                     Victoria Village  
2                            Regent Park, Harbourfront  
3                     Lawrence Manor, Lawrence Heights  
4                        Ontario Provincial Government  
..                                                 ...  
98       The Kingsway, Montgomery Road, Old Mill North  
99                                Church and Wellesley  
100                                     Enclave of M4L  
101  Old Mill South, King's Mill Park, Sunnylea, Hu...  
102  Mimico NW, The Queensway West, South of Bloor,...  

[103 rows x 3 columns]

*No postalcode is repeating, all are unique, out of 103, there is only 15 Borough is unqiue
<br>
*Most common Borough is North York and repeated 24 times, that means it has more Postalcode than any other place


In [66]:
df.describe()

PostalCode     Borough                 Neighborhood
count         103         103                          103
unique        103          15                          103
top           M2N  North York  St. James Town, Cabbagetown
freq            1          24                            1

*to clean Data i am using here dropna() built-in-function <br>
*Pandas dropna() method allows the user to analyze and drop Rows/Columns with Null values <br>
*Drop None rows of df and row which contains 'Not assigned' value <br>
*All "Not assigned" will be replace to 'NaN' <br>

In [68]:
df = df.dropna()
empty = 'Not assigned'
df = df[(df.PostalCode != empty ) & (df.Borough != empty) & (df.Neighborhood != empty)]

at last we will see the count of row and column by using shape

In [72]:
df.shape

(103, 3)

Here we can see 103 rows and 3 columns in the dataframe